flask/app/datasets.py

```diff
@@ -138,3 +138,65 @@ def update_dataset(id: int):
     routes.transaction_or_abort(db.session.commit)
 
     return jsonify(table)
+
+
+@app.route("/api/datasets", methods=["POST"])
+@login_required
+def post_dataset():
+    if not request.json:
+        return "Request body must be JSON", 400
+
+    dataset_type = request.json.get("dataset_type")
+    if not dataset_type:
+        return "A dataset type must be provided", 400
+
+    tissue_sample_id = request.json.get("tissue_sample_id")
+    if not tissue_sample_id:
+        return "A tissue sample id must be provided", 400
+
+    models.TissueSample.query.filter_by(
+        tissue_sample_id=tissue_sample_id
+    ).first_or_404()
+
+    col_names = [key for key in request.json]
```

Comment:
This will check all keys in the json regardless of whether they are valid and can be added under a dataset. I think these should be checking a pre-defined list of keys similar to the other POST/PATCH endpoints - what do you think @kevinlul

Reply:
good point! i think that makes sense- hadn't considered that it would be checking invalid keys